source/puppet/4.1/config_ssl_external_ca.markdown

```diff
@@ -201,230 +107,70 @@ Credential                        | File location
-      |    Master CA    |                |    Agent CA    |
-      |                 |                |                |
-      +--------+--------+                +--------+-------+
-               |                                  |
-               |                                  |
-               v                                  v
-      +-----------------+                +----------------+
-      |                 |                |                |
-      | Master SSL Cert |                | Agent SSL Cert |
-      |                 |                |                |
-      +-----------------+                +----------------+
-
-In this configuration, Puppet agents are configured to only authenticate peer certificates issued by the Master CA. Puppet masters are configured to only authenticate peer certificates issued by the Agent CA.
+## Option 2: Puppet master functioning as an intermediate CA
 
-> **Note:** If you're using this configuration, you can't use the ActiveRecord inventory service backend with multiple Puppet master servers. Use [PuppetDB][] for the inventory service instead.
+The puppet master can operate as an intermediate CA to an external Root CA.  The puppet master cannot be an intermediate to an intermediate.  In this mode the puppet master CA is left enabled and generation of agent certificates can remain automated, however there are some limitations:
 
-### Puppet Master
+* Agent-side CRL checking is not possible however CRL verification will still happen on the puppetserver
+* The CA certificate bundle (ie the external Root CA combined with the Intermediate CA certificate) must be distributed to the agents manually - ideally before puppet runs
 
-{{ master_basic }}
+### Puppet master
 
-You must also create a **CA bundle** for the web server. Append the **Agent CA certificate and Root CA certificate** together; the Root CA certificate must be located after the Agent CA certificate within the file.
+Before configuring the puppet master, you will need to obtain the intermediate CA certificate from your external Root CA.  Generating the Intermediate CA cert is outside the scope of the doc, since it will depend your external certificate authority solution.  This guide assumes you are either starting with a fresh installation or have removed all SSL files from your existing master and are starting over.  Also, you should stop all puppet related services on the master server before this process.
 
-~~~ bash
-cat agent_ca.pem root_ca.pem > ca_bundle_for_master.pem
-~~~
+In order to configure the puppet master you will need the following files placed:
 
-Put this file somewhere predictable. Puppet doesn't use it directly, but it can live alongside Puppet's copies of the certificates.
+Certificate     | Purpose                     | File Location
+----------------|--------------------------------------------
+ca_crt.pem      | Intermediate CA Certificate | /etc/puppetlabs/puppet/ssl/ca/ca_crt.pem
```

Comment:
These paths seem suspect to be because `/etc/puppetlabs/puppet/ssl/ca` (ostensibly) should only be present on Puppet masters that are also CAs. Since Puppetserver can't operate as a CA can these files live in `/etc/puppetlabs/puppet/ssl/*` ? For instance can we keep the certs in `/etc/puppetlabs/puppet/ssl/certs`?